src/crawlee/parsel_crawler/_parsel_crawling_context.py

```diff
@@ -1,26 +1,21 @@
-from __future__ import annotations
+from dataclasses import dataclass, fields
 
-from dataclasses import dataclass
-from typing import TYPE_CHECKING
+from parsel import Selector
+from typing_extensions import Self
 
-from crawlee._types import BasicCrawlingContext, EnqueueLinksFunction
 from crawlee._utils.docs import docs_group
-from crawlee.http_crawler import HttpCrawlingResult
-
-if TYPE_CHECKING:
-    from parsel import Selector
+from crawlee.abstract_http_crawler._http_crawling_context import ParsedHttpCrawlingContext
 
 
 @dataclass(frozen=True)
 @docs_group('Data structures')
-class ParselCrawlingContext(HttpCrawlingResult, BasicCrawlingContext):
-    """The crawling context used by the `ParselCrawler`.
-
-    It provides access to key objects as well as utility functions for handling crawling tasks.
-    """
-
-    selector: Selector
-    """The Parsel `Selector` object for the current page."""
-
-    enqueue_links: EnqueueLinksFunction
-    """The Parsel `EnqueueLinksFunction` implementation."""
+class ParselCrawlingContext(ParsedHttpCrawlingContext[Selector]):
```

Comment:
Please add some class docstring, because of the API doc renders. Every public should have it.